### PR TITLE
sftpgo: Support Topology Spread Constraints

### PR DESCRIPTION
**Pull request by stephan2012**
This PR adds support for [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)

Signed-off-by: Stephan Austermühle <au@hcsd.de>

### CODE REVIEW

File: charts/sftpgo/README.md [hunk 1]
```diff
@@ -142,3 +142,7 @@ require at least one port.
...
 | affinity | object | `{}` | [Affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity) configuration. See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#scheduling) for details. |
 | hostNetwork | bool | `false` | Run pods in the host network of nodes. Warning: The use of host network is [discouraged](https://kubernetes.io/docs/concepts/configuration/overview/#services). Make sure to use it only when absolutely necessary. |
+| topologySpreadConstraints.enabled | bool | `false` | Enable Kubernetes [Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) |
+| topologySpreadConstraints.maxSkew | int | `1` | Degree to which Pods may be unevenly distributed |
+| topologySpreadConstraints.topologyKey | string | `topology.kubernetes.io/zone` | Topology key |
+| topologySpreadConstraints.whenUnsatisfiable | string | `DoNotSchedule` | How to deal with a Pod if it doesn't satisfy the spread constraint |
```
Comment (sagikazarmark) [inline charts/sftpgo/README.md:148]: The readme is auto generated. Please add these comments to the values file and run `make docs`.

File: charts/sftpgo/templates/deployment.yaml [hunk 2]
```diff
@@ -133,3 +133,14 @@ spec:
...
       {{- end }}
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- if .Values.topologySpreadConstraints.enabled }}
+      topologySpreadConstraints:
+      - labelSelector:
+          matchLabels:
+            {{- include "sftpgo.selectorLabels" . | nindent 12 }}
+        maxSkew: {{ .Values.topologySpreadConstraints.maxSkew }}
+        topologyKey: {{ .Values.topologySpreadConstraints.topologyKey }}
+        whenUnsatisfiable: {{ .Values.topologySpreadConstraints.whenUnsatisfiable }}
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/deployment.yaml:144]: Please add one more level of indentation.

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution @stephan2012 ! Please see my comments on the PR. Other than that, LGTM.

File: charts/sftpgo/templates/deployment.yaml (hunk 2 above)
Comment (stephan2012) [reply charts/sftpgo/templates/deployment.yaml]: Done.

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**